modules/logging/src/test/java/io/oasp/module/logging/common/impl/SecureLoggingLogbackTest.java

```diff
@@ -0,0 +1,248 @@
+    // assertTrue("this pops up at the end of this test (with masking).", false);
+  }
+
+  /**
+   * Test the ExcludeClassifiedMarkerFilter on an event with MultiMarker (other tests are done within OWASP).
+   */
+  @Test
+  public void testExclClassifMarkerFilter() {
+
+    Marker marker = SecureLogging.SECURITY_SUCCESS_CONFIDENTIAL;
+
+    this.loggerLogback.info(marker, "confidential security message with MultiMarker.");
+
+    // Now verify our logging interactions
+    verify(this.mockAppender).doAppend(this.captorLoggingEvent.capture());
+
+    // Get the logging event from the captor
+    final LoggingEvent loggingEvent = this.captorLoggingEvent.getValue();
+
+    // Check log level is correct
+    assertThat(loggingEvent.getLevel()).isEqualTo(Level.INFO);
+
+    // Check the stand-alone filter decision for this event
+    assertEquals(FilterReply.DENY, this.filterExclClassif.decide(loggingEvent));
+
+    // Check the filter chain decision for this event (does not work)
+    // assertEquals(FilterReply.DENY, this.mockAppender.getFilterChainDecision(loggingEvent));
+  }
+
+  /**
+   * Test if a combined Marker contains the names of its constituent Markers. This test is useful in particular if the
+   * dependency org.owasp is not available, but also works when it is present. To test the fall back solution in
+   * {@link SecureLogging}, one has to create a separate logging module strictly without the OWASP dependency.
+   */
+  @Test
+  public void testInitMarkersByName() {
+
+    // setup: SecureLogging.initMarkers() is called by the loc below.
+    Marker multiMarker = SecureLogging.SECURITY_SUCCESS_CONFIDENTIAL;
+    Marker securMarker = SecureLogging.SECURITY_SUCCESS;
+    Marker confidMarker = SecureLogging.CONFIDENTIAL;
+
+    // verify that the combined Marker or MultiMarker contains the names of its constituent Markers.
+    assertFalse("name of combined marker must not be empty.", multiMarker.getName().isEmpty());
+    assertTrue("name of combined marker must contain the names of its constituent Markers.",
+        multiMarker.getName().contains(securMarker.getName()));
+    assertTrue("name of combined marker must contain the names of its constituent Markers.",
+        multiMarker.getName().contains(confidMarker.getName()));
+  }
+
+  /**
+   * Test Marker creation if the dependency org.owasp is available, which provides the class
+   * {@link org.owasp.security.logging.MultiMarker}.
+   */
+  @Test
+  public void testInitWithMultiMarkerClass() {
+
+    // skip test if the dependency is not available.
+    if (!SecureLogging.hasExtClass())
+      return;
+    assertTrue("this test fails if the dependency org.owasp is not available.", SecureLogging.hasExtClass());
+
+    // setup
+    // SecureLogging.initMarkers() is called by the loc below:
+    Marker multiMarker = SecureLogging.SECURITY_SUCCESS_CONFIDENTIAL;
+    Marker securMarker = SecureLogging.SECURITY_SUCCESS;
+    Marker confidMarker = SecureLogging.CONFIDENTIAL;
```

Comment:
IMHO making these to local variables is pointless and even contradictory for readability of asserts.